innopoints/models/activity.py

```diff
@@ -78,6 +78,11 @@ def vacant_spots(self):
                                                status=ApplicationStatus.approved).count()
         return max(self.people_required - accepted, -1)
 
+    def has_application_from(self, user):
+        """Return whether the given user has applied for this activity."""
+        application = Application.query.filter_by(applicant=user, activity_id=self.id)
+        return db.session.query(application.exists()).scalar()
```

Comment:
Assuming that `.scalar()` returns 0 or 1 (I didn't run the code and it's not clear from the docs), shouldn't you cast it to bool (as the function name suggests it returns)?

Reply:
No, it returns a boolean (as the SO answers suggest)